analyzers/haskell/analyzer/stack.go

```diff
@@ -0,0 +1,70 @@
+package analyzer
+
+import (
+	"github.com/fossas/fossa-cli/buildtools/stack"
+	"github.com/fossas/fossa-cli/graph"
+	"github.com/fossas/fossa-cli/module"
+	"github.com/fossas/fossa-cli/pkg"
+)
+
+func AnalyzeStack(m module.Module) (graph.Deps, error) {
+	stackAllDeps, err := stack.GetStackDependencies(m.Dir, nil)
+
+	if err != nil {
+		return graph.Deps{}, err
+	}
+
+	ghcPkgDeps, err := stack.GetGhcPkgDepMap()
+
+	if err != nil {
+		return graph.Deps{}, err
+	}
+
+	depth := int(1)
+	stackImmediateDeps, err := stack.GetStackDependencies(m.Dir, &depth)
+
+	if err != nil {
+		return graph.Deps{}, err
+	}
+
+	return AnalyzeStackPure(stackAllDeps, stackImmediateDeps, ghcPkgDeps), nil
```

Comment:
Is this logic we could reasonably contain in the build tool? I would prefer doing it that way so that the analyzer really only thinks about strategy and discovery.

Reply:
I can move this function to the build tool if that's a better place. I feel strongly that the "pure" code (deterministic based on input, and cannot perform effects -- including failure) should be separate from the effectful code when possible. It makes the `AnalyzeStackPure` function trivial to test and reason about